lang/src/jmh/kotlin/org/partiql/lang/PartiQLBenchmark.kt

```diff
@@ -0,0 +1,163 @@
+     * clean up ("tear down") the state object after the benchmark has been executed.
+     *
+     * The setup and tear down execution time is not included in the benchmark runtime measurements.
+     */
+    @State(Scope.Thread)
+    open class MyState2 {
+        @Setup(Level.Trial)
+        fun doSetup() {
+            sum = 0
+            println("Do Setup")
+        }
+
+        @TearDown(Level.Trial)
+        fun doTearDown() {
+            println("Do TearDown")
+        }
+
+        var a = 1
+        var b = 2
+        var sum = 0
+
+        val ion = IonSystemBuilder.standard().build()
+        val parser = SqlParser(ion)
+        val pipeline = CompilerPipeline.standard(ion)
+
+        val data = """
+            { 
+                'hr': { 
+                    'employeesNestScalars': <<
+                        { 
+                            'id': 3, 
+                            'name': 'Bob Smith', 
+                            'title': null, 
+                            'projects': [ 
+                                'AWS Redshift Spectrum querying',
+                                'AWS Redshift security',
+                                'AWS Aurora security'
+                            ]
+                        },
+                        { 
+                            'id': 4, 
+                            'name': 'Susan Smith', 
+                            'title': 'Dev Mgr', 
+                            'projects': []
+                        },
+                        { 
+                            'id': 6, 
+                            'name': 'Jane Smith', 
+                            'title': 'Software Eng 2', 
+                            'projects': [ 'AWS Redshift security' ]
+                        }
+                    >>
+                } 
+            }
+        """.trimIndent()
+        val bindings = pipeline.compile(parser.parseExprNode(data)).eval(EvaluationSession.standard()).bindings
+        val session = EvaluationSession.build { globals(bindings) }
+
+        val query = "SELECT * FROM hr.employeesNestScalars"
+        val exprNode = parser.parseExprNode(query)
+        val expression = pipeline.compile(exprNode)
+    }
+
+    /**
+     * You can use different [BenchmarkMode]. The benchmark mode tells JMH what you want to measure.
+     * [TimeUnit] specifies what time units you want the benchmark results printed in.
+     */
+    @Benchmark
+    @BenchmarkMode(Mode.AverageTime)
+    @OutputTimeUnit(TimeUnit.MILLISECONDS)
+    fun testMethod2(state: MyState2) {
+        state.sum = state.a + state.b
+    }
+
+    /**
+     * Pass the final values to the [Blackhole] to fool JMH that it is indeed been used.
+     */
+    @Benchmark
+    fun testMethod3(state: MyState, blackhole: Blackhole) {
+        val sum1 = state.a + state.b
+        val sum2 = state.a + state.a + state.b + state.b
+        blackhole.consume(sum1)
+        blackhole.consume(sum2)
+    }
```

Comment:
I'm sure this was helpful for getting it working on your machine but it doesn't belong in `main`.